DataManagementSystem/Client/DataManager.py

```diff
@@ -904,14 +904,17 @@ def registerFile( self, fileTuple, catalog = '' ):
     :param tuple fileTuple: (lfn, physicalFile, fileSize, storageElementName, fileGuid, checksum )
     :param str catalog: catalog name
     """
-    if isinstance( fileTuple, list ):
-      fileTuples = fileTuple
-    elif isinstance( fileTuple, tuple ):
-      fileTuples = [fileTuple]
+    if isinstance( fileTuple, ( list, dict, set, tuple ) ):
+      fileTuples = list( fileTuple )
     else:
-      errStr = "registerFile: Supplied file info must be tuple of list of tuples."
-      self.log.debug( errStr )
-      return S_ERROR( errStr )
+      fileTuples = [fileTuple]
+    for fileTuple in fileTuples:
+      if not isinstance( fileTuple, tuple ):
+        errStr = "registerFile: Supplied file info must be tuple or list of tuples."
+        self.log.debug( errStr )
+        return S_ERROR( errStr )
+    if not fileTuples:
+      return S_OK( {'Successful':[], 'Failed':{}} )
```

Comment:
  Yes, it is mandatory that in this
      structure both Successful and Failed are dicts.
      As a matter of definition.
      
        Cheers,
        Andrei
      
      On 09/06/15 12:58, Philippe Charpentier wrote:
    
    
      In DataManagementSystem/Client/DataManager.py:
      >      else:
> -      errStr = "registerFile: Supplied file info must be tuple of list of tuples."
> -      self.log.debug( errStr )
> -      return S_ERROR( errStr )
> +      fileTuples = [fileTuple]
> +    for fileTuple in fileTuples:
> +      if not isinstance( fileTuple, tuple ):
> +        errStr = "registerFile: Supplied file info must be tuple or list of tuples."
> +        self.log.debug( errStr )
> +        return S_ERROR( errStr )
> +    if not fileTuples:
> +      return S_OK( {'Successful':[], 'Failed':{}} )

      Well, this depends... Usually the test done after that is like:
      if lfn in result['Value']['Successful']
      of course if it is 
      for lfn, se in result['Value']['Successful'].items():
      there may be a problem... Probably {} is safer in this respect,
        right as most (all?) constructs possible when expecting a list
        would work with a dict?
      —
        Reply to this email directly or view
          it on GitHub.